lib/rules/unit-no-unknown/index.js

```diff
@@ -72,9 +72,13 @@ const rule = function(actual, options) {
       });
     }
 
-    root.walkAtRules(/^media$/i, atRule =>
-      check(atRule, atRule.params, atRuleParamIndex)
-    );
+    root.walkAtRules(atRule => {
+      if (!/^media$/i.test(atRule.name) && !atRule.variable) {
```

Comment:
Same as above, there were a bunch of tests for `@variable` syntax, hence this change even though it's non-standard syntax. Perhaps those tests should be updated instead?